go/vt/sqlparser/ast.go

```diff
@@ -499,22 +499,25 @@ func (node *Set) WalkSubtree(visit Visit) error {
 }
 
 // DDL represents a CREATE, ALTER, DROP or RENAME statement.
-// Table is set for AlterStr, DropStr, RenameStr.
+// Table is set for AlterStr, DropStr, RenameStr, ReorganizeStr
 // NewName is set for AlterStr, CreateStr, RenameStr.
 type DDL struct {
-	Action    string
-	Table     TableName
-	NewName   TableName
-	IfExists  bool
-	TableSpec *TableSpec
+	Action               string
+	Table                TableName
+	NewName              TableName
+	IfExists             bool
+	TableSpec            *TableSpec
+	PartitionName        ColIdent
```

Comment:
It should be a relatively simple move around of a couple lines of code to make this new construct be in line with `*TableSpec`. Not sure why we can't do that just because in the future we may or may not want to make legit interfaces for everything.  I think we should start with separate fields for each in the DDL and if it becomes too much down the line we can split it up, and if not then not.